src/Microsoft.DotNet.Build.Tasks.Feed/MergeBlobs.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using MSBuild = Microsoft.Build.Utilities;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.DotNet.Build.Tasks.Feed
+{
+    public sealed class MergeBlobs : MSBuild.Task
+    {
+        [Required]
+        public string BaseOutputUrl { get; set; }
+
+        [Required]
+        public string AccountKey { get; set; }
+
+        [Required]
+        public ITaskItem[] BaseInputUrls { get; set; }
+
+        [Required]
+        public ITaskItem[] FoldersToMerge { get; set; }
+
+        public bool Overwrite { get; set; }
+
+        public bool SkipCreateContainer { get; set; } = false;
+
+        public bool SkipIfMissing { get; set; } = true;
+
+        public override bool Execute()
+        {
+            return ExecuteAsync().GetAwaiter().GetResult();
+            
+        }
+
+        private string GetCanonicalStorageUri(string uri, string subPath)
+        {
+            string newUri = uri.TrimEnd('/');
+            if (!string.IsNullOrEmpty(subPath))
+            {
+                newUri = $"{newUri}/{subPath.TrimStart('/')}";
+            }
+            return newUri;
+        }
+
+        public async Task<bool> ExecuteAsync()
+        {
+            try
+            {
+                Log.LogMessage(MessageImportance.High, "Performing blob merge...");
+
+                if (BaseInputUrls == null || FoldersToMerge == null)
+                {
+                    Log.LogError($"No items to merge. Please check ItemGroup ItemsToMerge.");
+                }
+                else
+                {
+                    await Task.WhenAll(FoldersToMerge.Select(async folder => {
+                        string folderName = folder.ItemSpec;
+                        // Canonicalize the target uri
+                        string targetUri = GetCanonicalStorageUri(BaseOutputUrl, folderName);
+                        // Invoke the blob URI parser on the target URI and deal with any container creation that needs to happen
+                        BlobUrlInfo targetUrlInfo = new BlobUrlInfo(targetUri);
+                        CloudStorageAccount storageAccount = new CloudStorageAccount(new WindowsAzure.Storage.Auth.StorageCredentials(targetUrlInfo.AccountName, AccountKey), true);
+                        CloudBlobClient client = storageAccount.CreateCloudBlobClient();
+                        CloudBlobContainer targetContainer = client.GetContainerReference(targetUrlInfo.ContainerName);
+
+                        if (!SkipCreateContainer)
+                        {
+                            Log.LogMessage($"Creating container {targetUrlInfo.ContainerName} if it doesn't exist.");
+                            await targetContainer.CreateIfNotExistsAsync();
+                        }
+                        
+                        // Copy blob to blob.
+                        await Task.WhenAll(BaseInputUrls.Select(async item => {
```

Comment:
This level of nested functions just begs for the methods to be split out. Right now it is kind of hard to tell which function has which context.

Reply:
@weshaggard I think I'm going to redo this a bit.  The intention here was to both avoid calling a utility build definition multiple times and avoid doing double batching in msbuild (which makes the build def pretty hard to read).  I think I'm just going to simplify this as a simple CopyBlobDirectory, then batch the source/targets in msbuild and call the utility build def as needed for each set of folders that will be copied.

As it is it's a simpler msbuild project but this task has less utlitity.